src/spatial/collision.js

```diff
@@ -289,11 +281,11 @@ Crafty.c("Collision", {
             id = obj[0];
 
             //check if not added to hash and that actually intersects
-            if (!dupes[id] && this[0] !== id && obj.__c[comp] && overlap(oarea, area))
+            if (!dupes[id] && this[0] !== id && obj.__c[component] && overlap(oarea, area))
                 dupes[id] = obj;
         }
 
-        for (key in dupes) {
+        for (key in dupes) if (dupes.hasOwnProperty(key)) {
```

Comment:
I added `hasOwnProperty` for each for/in, just in case. If you think it's no need for that - just tell me, i'll revert.